### PR TITLE
Introduce table options

### PR DESCRIPTION
**Pull request by theopolis**
This pushed table spec information into the class `TablePlugin` definition so we can make decisions and alert query-callsites when they use a table incorrectly.

We can now write error messages if a table, like `hash` or `file`, expects a `path` parameter in the predicate and one is not provided.

### CODE REVIEW

**Comment by osqueryer** [comment]
Build origin/pr/2101/merge (Job: 2678) failed :-1:

**Comment by osqueryer** [comment]
Build origin/pr/2101/merge (Job: 2679) is good! :+1:

**Comment by osqueryer** [comment]
Build origin/pr/2101/merge (Job: 2680) is good! :+1:

**Comment by theopolis** [comment]
I am labeled this an API change since it begins to change the schema communicated over the `Registry::call` response for table columns.

File: osquery/core/tables.cpp [hunk 1]
```diff
@@ -126,8 +126,9 @@ Status TablePlugin::call(const PluginRequest& request,
     // information such as name and type.
     const auto& column_list = columns();
     for (const auto& column : column_list) {
-      response.push_back(
-          {{"name", column.first}, {"type", columnTypeName(column.second)}});
+      response.push_back({{"name", std::get<0>(column)},
+                          {"type", columnTypeName(std::get<1>(column))},
+                          {"op", INTEGER(std::get<2>(column))}});
```
Comment (theopolis) [inline osquery/core/tables.cpp]: This is an API change.

File: osquery/core/tables.cpp [hunk 2]
```diff
@@ -146,8 +147,9 @@ PluginResponse TablePlugin::routeInfo() const {
   // Route info consists of only the serialized column information.
   PluginResponse response;
   for (const auto& column : columns()) {
-    response.push_back(
-        {{"name", column.first}, {"type", columnTypeName(column.second)}});
+    response.push_back({{"name", std::get<0>(column)},
+                        {"type", columnTypeName(std::get<1>(column))},
+                        {"op", INTEGER(std::get<2>(column))}});
```
Comment (theopolis) [inline osquery/core/tables.cpp]: This is an API change.

**Pull request merged**